neqo-transport/src/connection.rs

```diff
@@ -506,6 +506,24 @@ impl Connection {
                 Some(ref t) => {
                     qtrace!("TLS token {}", hex(&t));
                     let mut enc = Encoder::default();
+                    let smoothed_rtt = match self.loss_recovery.smoothed_rtt() {
+                        Some(v) => v,
+                        None => {
+                            debug_assert!(false, "smoothed_rtt should not be None");
+                            return None;
+                        }
+                    };
+                    let smoothed_rtt = match u32::try_from(smoothed_rtt.as_millis()) {
+                        Ok(smoothed_rtt) => smoothed_rtt,
+                        _ => {
+                            debug_assert!(
+                                false,
+                                "failed to cast smoothed_rtt value from ms to u32"
+                            );
+                            return None;
+                        }
+                    };
+                    enc.encode_uint(4, smoothed_rtt);
```

Comment:
Thanks for the explanation. I've changed it. 

Initially I was not sure whether setting 0 as default in case if try_from fails or if smoothed_rtt is None is okay. So Dragana suggested to return None in such cases and add debug_asserts. Thus I ended up wrapping those into explicit matches.